internal/certgen/certgen.go

```diff
@@ -196,7 +288,10 @@ func OutputCerts(certgenConfig *Config, certs *ContourCerts) error {
 		}
 
 		fmt.Printf("Outputting certs to PEM files in %s/\n", certgenConfig.OutputDir)
-		certs.writeCertPEMs(certgenConfig.OutputDir)
+		err = certs.writeCertPEMs(certgenConfig.OutputDir)
```

Comment:
As an added bonus, if you move GenerateCerts into cmd/contour you can use the check helper 'cos its totally fine for cmd/contour to blow up on error whereas its not cool for a library to completely dump the program its called from.

Then that would give you an easy way to call `check(os.MkdirAll(certgenConfig.OutputDir)) and know that the output directory existed beforehand.